src/StatsBase.jl

```diff
@@ -215,6 +215,24 @@ export
     ZScoreTransform,       # the type to represent a z-score data transformation
     UnitRangeTransform     # the type to represent a 0-1 data transformation
 
+function __init__()
+    @require CUDA="052768ef-5323-5732-b1bb-66c8b64840ba" begin
+        using .CUDA
+        function _compute_extrema(X::Union{CuMatrix{<:Real},Adjoint{T,CuMatrix{T}} where T<:Real})
```

Comment:
It is used [here](https://github.com/JuliaStats/StatsBase.jl/blob/1c3f3d7ae4fd8afae126443d15ddaf8029316618/src/transformations.jl#L274) in the `fit(::Type{UnitRangeTransform}, ...)` function. The existing implementation does not work (or, is very slow) for `CuArray`s because the array is indexed.

Reply:
Sorry, but I don't think using Requires is OK for a package like StatsBase, as it will increase load times of many packages. We should find better abstractions so that StatsBase doesn't have to be aware of CUDA.

Couldn't `_compute_extrema` be replaced with `extrema.(eachcol(X))`? Would that be enough to use optimized CUDA methods?

Reply:
I was not too happy with using Requires either. I have removed it now. I also replaced `_compute_extrema` by `extrema`. The caveat that `extrema` has not yet been implemented for CUDA arrays, but it is reasonable to expect that it will become available in the future; it is mentioned in https://github.com/JuliaGPU/CUDA.jl/issues/265#issue-649177087